### DISCUSSION TITLE
Rendering a stl file

### DISCUSSION THREAD

**Discussion by tharakanandakumara** [StereoKit/StereoKit#152] [Q&A]
I tried  to import a stl file using Model.FromFile method. but it doesnot show anything after rendered. stl is in ASCII format.

**Comment by maluoi** [comment]
Ok, thanks for letting me know! I've got a few questions:

- Do you see anything related to this in the logs? There's usually a note if the file couldn't be found or read.
- Does Model.FromFile return null, or is it still providing a valid Model object? A valid Model object would indicate a successful parsing of the .stl file.
- Does your .stl's scale match StereoKit's 1 unit = 1 meter? Objects can often have a wild variety of different sizes, and it's not uncommon to find yourself inside of a _very_ large 3D model, which would prevent you from seeing it.

Also, if it's possible for you to share a link to an .stl where this problem happens, it would help a lot for debugging!

**Reply by tharakanandakumara** [reply]
Thanks for replying.I can see a Model object has been parsed when it's loaded.
I have attached my stl file here.

**Reply by tharakanandakumara** [reply]
[tablenew.zip](https://github.com/maluoi/StereoKit/files/6563267/tablenew.zip)

**Reply by maluoi** [reply]
Okay, thanks for the file! The model loaded just fine, it's definitely a scale issue. The model here is about 4.6km by 1.8km by 20m, so you're almost certainly seeing it from inside of the mesh! You'll either want to scale this down in your modeling tool, or scale it down when creating a transform matrix for it.

A nice trick for just ensuring a 'normalized' visible scale is to do something like this:
https://github.com/maluoi/StereoKit/blob/master/Examples/StereoKitTest/Demos/DemoPicker.cs#L65

**Reply by maluoi** [reply]
The STL format unfortunately doesn't have any established standards about the coordinate system, so this is a limitation of the file format.

I may look into adding coordinate system conversion options when loading a file, to make some of this stuff easier.
